pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -1471,11 +1823,377 @@ def add_lift_test_measurements(
+
+        Parameters
+        ----------
+        samples : az.InferenceData
+            The inference data containing the channel contributions.
+        figsize : tuple[float, float], optional
+            The size of the figure to be created, by default (12, 6).
+        original_scale : bool, optional
+            A boolean flag to determine if the values should be plotted in their original scale,
+            by default True.
+
+        Returns
+        -------
+        tuple
+            A tuple containing the matplotlib figure and axes of the plot.
+        """
+
+        # def plot_budget_allocation(self, samples, ax, original_scale=True):
+        if original_scale:
+            channel_contributions = (
+                samples["channel_contributions"]
+                .mean(dim=["sample"])
+                .mean(dim=["date"])
+                .values
+                * self.get_target_transformer()["scaler"].scale_
+            )
+
+            allocate_spend = (
+                np.array(list(self.optimal_allocation_dict.values()))
+                * self.channel_transformer["scaler"].scale_
+            )
+
+        else:
+            channel_contributions = (
+                samples["channel_contributions"]
+                .mean(dim=["sample"])
+                .mean(dim=["date"])
+                .values
+            )
+            allocate_spend = np.array(list(self.optimal_allocation_dict.values()))
+
+        if ax is None:
+            fig, ax = plt.subplots(figsize=figsize)
+        else:
+            fig = ax.get_figure()
+
+        bar_width = 0.35
+        opacity = 0.7
+
+        index = np.arange(len(self.channel_columns))
+
+        bars1 = ax.bar(
+            index,
+            allocate_spend,
+            bar_width,
+            color="b",
+            alpha=opacity,
+            label="Allocate Spend",
+        )
+
+        ax2 = ax.twinx()
+
+        bars2 = ax2.bar(
+            index + bar_width,
+            channel_contributions,
+            bar_width,
+            color="r",
+            alpha=opacity,
+            label="Channel Contributions",
+        )
+
+        ax.set_xlabel("Channels")
+        ax.set_ylabel("Allocate Spend", color="b")
+        ax.tick_params(axis="x", rotation=90)
+        ax.set_xticks(index + bar_width / 2)
+        ax.set_xticklabels(self.channel_columns)
+
+        ax.set_ylabel("Allocate Spend", color="b", labelpad=10)
+        ax2.set_ylabel("Channel Contributions", color="r", labelpad=10)
+
+        # Remove grid lines
+        ax.grid(False)
+        ax2.grid(False)
+
+        # Add legend
+        bars = [bars1[0], bars2[0]]
+        labels = [bar.get_label() for bar in bars]
+        ax.legend(bars, labels)
+
+        return fig, ax
+
+    def plot_allocated_contribution_by_channel(
+        self,
+        samples: az.InferenceData,
+        lower_quantile: float = 0.025,
+        upper_quantile: float = 0.975,
+        original_scale: bool = True,
+    ):
```

Comment:
as above: add type hint, add it to docstrings and return just figure

Reply:
Done